docs: add charm docs exemplars links (#2253)
canonical/charmcraft: docs/howto/manage-charms.rst

Add docs
~~~~~~~~

> When you publish your charm on Charmhub, reference documentation about the charm's

Reviewer: > When

I have a feeling that the "if" was there to suggest that publication to Charmhub wasn't mandatory. We usually keep private stores in mind.

Reviewer 2: Or indeed charms that never go into a store and just get deployed through CI/CD

Author: Reverted the changes back to "if".

resources, actions, configurations, relations, and libraries is generated and 
published automatically in respective tabs.

> Use the Description tab to add additional documentation, such as tutorials or 

Reviewer: ```suggestion

On the **Description** tab, add additional documentation, such as tutorials or 
```

I think this would read better with a paragraph break here.

Author: Fixed in https://github.com/canonical/charmcraft/pull/2253/commits/8ef59e0da29fd0535a9d45fe03fa8ec90af3ef13

how-to guides, focusing specifically on the charm itself. 
For workload-specific or Juju-related content, 
link to the appropriate upstream documentation.

To add content to the Description tab, create a Discourse topic and include its URL 
in your charm's project file under the 

> :ref:`links.documentation <charmcraft-yaml-key-documentation>` key:

Reviewer: Keep in mind I've never managed a charm, but I feel like these two paragraphs step on each other. Both describe adding content on the tab but in different ways. Should they be combined?

Reviewer 2: Agreed - combined witch your previous suggestion of putting this in a new paragraph that flows better IMO.

Author: I've experimented with it a bit. Please check the [new version](https://github.com/canonical/charmcraft/pull/2253/commits/8ef59e0da29fd0535a9d45fe03fa8ec90af3ef13)

~~~~~~~~

If you publish your charm on Charmhub, reference documentation about the charm's
resources, actions, configurations, relations, and libraries is generated and 
published automatically in respective tabs.

> To add content to the Description tab, 

Reviewer: Any chance we can use bold to fit with the recommended UI formatting? https://docs.ubuntu.com/styleguide/en/#bold-vs.-italics

    See more: :ref:`charmcraft-yaml-key-links`

The Description tab can contain additional documentation, such as tutorials or 
how-to guides.

> .. note::

Reviewer: I like the update! I think it flows well.

Do you think we need the note block? I think it could work as a regular paragraph.

Author: Yes, this fragment is important enough to use the note block.
Given that we are in a How-to guide, it feels too generic to put it in the middle of more practical instructions as just usual text.

   A charm's documentation should focus on the charm itself.
   For workload-specific or Juju-related content, link to the appropriate upstream 
   documentation.

> A smaller charm can have a single-page documentation for its description. 
> A bigger charm, that needs a multi-page documentation, can have either

Reviewer: ```suggestion
A smaller charm can have a single-page document for its description. 
A bigger charm, that needs multi-page documentation, can have either
```

Grammar-related.

Author: Yes, we can delete the articles, but I don't think the other word has the same meaning. We really want to talk about charm's documentation that fits into a single web page.

```suggestion
A smaller charm can have single-page documentation for its description. 
A bigger charm, that needs multi-page documentation, can have either
```

    See more: :ref:`charmcraft-yaml-key-links`

> Smaller charms can have a single-page documentation in the Description tab. 
> Bigger charms with multi-page documentation, should have a full
> [Diátaxis](https://diataxis.fr/) navigation tree in the Description tab, 
> or just provide a brief description with a link to an external documentation set. 

Reviewer: ```suggestion
A smaller charm can have a single-page documentation for its description. 
A bigger charm, with multi-page documentation, should have a full
[Diátaxis](https://diataxis.fr/) navigation tree in its description, 
or just provide a brief description with a link to an external documentation set. 
```

I think we can reduce how much we speak about the tab itself.

Author: I don't think we can replace all mentions of the tab, since we need to refer to the exact UI element sometimes, but I've adjusted the text to minimise the usage. Up to one per paragraph.

Reviewer: Looks great!

.. admonition:: Best practice
    :class: hint

>     Smaller charm documentation exemplars:

Reviewer: > exemplar

This term came up before in a discussion about user-facing docs. Do you think for ease of understanding we should use "example" instead?

Author: Yes, I see how the word example can be easier to understand. Although the meaning is not precisely the same, I guess in this context we can replace it. Fixed in https://github.com/canonical/charmcraft/pull/2253/commits/8ef59e0da29fd0535a9d45fe03fa8ec90af3ef13
